routes/user.js

```diff
@@ -0,0 +1,238 @@
+        var password = validator.toString(validator.escape(req.body.password));
+        var email    = validator.toString(validator.escape(req.body.email));
+
+        // Captcha key
+
+        /////////////////////
+        // USERNAME CHECKS //
+        /////////////////////
+
+        // Checks empty username field
+        if (!username) {
+            res.send({
+                code    : 400,
+                message : 'Username field is required.'
+            });
+            return;
+        }
+
+        // Validate username (alphanumeric characters, underscore and/or dashes)
+        if (!isUsername(username)) {
+            res.send({
+                code    : 400,
+                message : 'Username can only contain alphanumeric characters, ' +
+                          'underscores, and/or dashes.'
+            });
+            return;
+        }
+
+        // TODO: Decide min and max
+        // Check username length
+        if (!validator.isLength(username, 4, 12)) {
+            res.send({
+                code    : 400,
+                message : 'Username must be between 4 and 12 characters.'
+            });
+            return;
+        }
+
+        /////////////////////
+        // PASSWORD CHECKS //
+        /////////////////////
+
+        // Checks empty password field
+        if (!password) {
+            res.send({
+                code    : 400,
+                message : 'Password field is required.'
+            });
+            return;
+        }
+
+        // Check is valid password (must contain at least one letter and number)
+        
+        // TODO: define min and max
+        // Check password length
+        if (!validator.isLength(password, 4, 512)) {
+            res.send({
+                code    : 400,
+                message : 'Password must be between 5 and 512 characters.'
+            });
+            return;
+        }
+
+        //////////////////
+        // EMAIL CHECKS //
+        //////////////////
+
+        // Validate email
+        if (!validator.isEmail(email)) {
+            res.send({
+                code    : 400,
+                message : 'Not a valid email address.'
+            });
+            return;
+        }
+
+        // TODO: define min and max
+        // Check email length
+        if (!validator.isLength(email, 4, 512)) {
+            res.send({
+                code    : 400,
+                message : 'Email must be between 4 and 512 characters long.'
+            });
+            return;
+        }
+        
+        User.create(username, password, email, function(err, session) {
+            if (err) {
+                res.send(err);
+                return;
+            }
+            res.send({
+                code    : 200,
+                message : 'Account successfully created.',
+                token   : session.token,
+                ttl     : session.createdAt.setDate(session.createdAt.getDate() + 7).toString()
```

Comment:
This should instead be `ttl: session.createdAt + 604800000 // 604800000 is 7D in milliseconds`